src/main/java/org/spongepowered/common/data/provider/entity/EntityData.java

```diff
@@ -169,6 +174,13 @@ public static void register(final DataProviderRegistrator registrator) {
                         .get(EntityBridge::bridge$getDisplayNameText)
                         .set(EntityBridge::bridge$setDisplayName)
                         .delete(h -> h.bridge$setDisplayName(null));
+
+        registrator.newDataStore(SpongeEntitySnapshot.class, SpongeEntityArchetype.class)
+                .dataStore(Keys.DISPLAY_NAME,
+                (dv, v) -> dv.set(Constants.Entity.CUSTOM_NAME, SpongeAdventure.json(v)),
+                dv -> dv.getString(Constants.Entity.CUSTOM_NAME).map(SpongeAdventure::json));
```

Comment:
If you're going to multiline, indent with some explicits, this looks weird otherwise.

```suggestion
        registrator.newDataStore(SpongeEntitySnapshot.class, SpongeEntityArchetype.class)
                .dataStore(
                    Keys.DISPLAY_NAME,
                    (dv, v) -> dv.set(Constants.Entity.CUSTOM_NAME, SpongeAdventure.json(v)),
                    dv -> dv.getString(Constants.Entity.CUSTOM_NAME).map(SpongeAdventure::json)
                );
```